assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -64,17 +68,30 @@ const usePersistedStateReducer = (): [State, Dispatch] => {
   }, [selectedRouteIds, ladderDirections, ladderCrowdingToggles])
 
   useEffect(() => {
-    if (routeTabsToPush && !routeTabsPushInProgress) {
+    if (routeTabsToPush[0] && !routeTabsPushInProgress) {
       dispatch(startingRouteTabsPush())
-      putRouteTabs(routeTabsToPush)
+      putRouteTabs(routeTabsToPush[0])
         .then((response) => {
           if (response.ok) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
+            dispatch(routeTabsPushComplete())
+          } else if (routeTabsPushRetriesLeft > 0) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetriesLeft - 1)
+            dispatch(retryRouteTabsPushIfNotOutdated(routeTabsToPush[0]))
+          } else {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
             dispatch(routeTabsPushComplete())
+          }
+        })
+        .catch(() => {
+          if (routeTabsPushRetriesLeft > 0) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetriesLeft - 1)
+            dispatch(retryRouteTabsPushIfNotOutdated(routeTabsToPush[0]))
           } else {
-            dispatch(retryRouteTabsPushIfNotOutdated(routeTabsToPush))
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
+            dispatch(routeTabsPushComplete())
```

Comment:
If the user does another change to their route tabs, it will attempt pushing again. Otherwise, the change will be lost. Unfortunately we don't really have designs for how to surface an error with some data like this not making it to the Skate backend, when and how we should prompt the user to retry, etc. It's something that we have [a ticket](https://app.asana.com/0/1152340551558956/1201335236855131/f) to look into.